flash/core/data/io/input_transform.py

```diff
@@ -1170,15 +1183,20 @@ def __str__(self) -> str:
         )
 
 
-def _create_collate_input_transform_processors(
-    input_transform: "InputTransform", callbacks: List[FlashCallback]
-) -> Tuple[_InputTransformProcessorV2, _InputTransformProcessorV2]:
-    """This utility is used to create the 2 `_InputTransformProcessorV2` objects which contain the transforms used
-    as the DataLoader `collate_fn` and the DataModule `on_after_batch_transfer` hook."""
+def __make_collates(input_transform: "InputTransform", on_device: bool, collate: Callable) -> Tuple[Callable, Callable]:
+    if on_device:
+        return input_transform._identity, collate
+    return collate, input_transform._identity
+
+
+def __configure_worker_and_device_collate_fn(
+    running_stage: RunningStage, input_transform: "InputTransform"
```

Comment:
Since it was in the same file, type reference might not load (happens in the case when we have circular imports, not this maybe). So I put them just to be safe.

Reply:
Oh, makes sense. I missed that. Thanks!